deploy/kustomize/deploy_kustomize.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+set -e
+# Setup Env Variables
+export TARGET_NAMESPACE="${TARGET_NAMESPACE:-kubevirt-hyperconverged}"
+export APP_REGISTRY="${APP_REGISTRY:-kubevirt-hyperconverged}"
+export CSC_SOURCE="${CSC_SOURCE:-hco-catalogsource-config}"
+export PACKAGE="${PACKAGE:-kubevirt-hyperconverged}"
+export MARKETPLACE_NAMESPACE="${MARKETPLACE_NAMESPACE:-openshift-marketplace}"
+export HCO_VERSION="${HCO_VERSION:-1.0.0}"
+export HCO_CHANNEL="${HCO_CHANNEL:-1.0.0}"
+export PRIVATE_REPO="${PRIVATE_REPO:-false}"
+export QUAY_USERNAME="${QUAY_USERNAME:-}"
+export QUAY_PASSWORD="${QUAY_PASSWORD:-}"
+export QUAY_TOKEN=""
+export AUTH_TOKEN=""
+
+# Get Quay token if private repo is chosen
+get_quay_token() {
+    token=$(curl -sH "Content-Type: application/json" -XPOST https://quay.io/cnr/api/v1/users/login -d '
+  {
+      "user": {
+          "username": "'"${QUAY_USERNAME}"'",
+          "password": "'"${QUAY_PASSWORD}"'"
+      }
+  }' | jq -r '.token')
+
+  if [ "$token" == "null" ]; then
+    echo Got invalid Token from Quay. Please check your credentials in QUAY_USERNAME and QUAY_PASSWORD.
+    exit 1
+  else
+    eval "$1='$token'";
+  fi
+}
+
+if [ "$PRIVATE_REPO" = 'true' ]; then
+  if [ -z "$QUAY_USERNAME" ] || [ -z "$QUAY_PASSWORD" ]; then
+    echo Please set both "QUAY_USERNAME" and "QUAY_PASSWORD";
+    exit 1;
+  fi
+  get_quay_token QUAY_TOKEN
+  AUTH_TOKEN=$(cat <<-END
+  authorizationToken:
+    secretName: quay-registry-${APP_REGISTRY}
+END
+)
+fi
+
+# Perform variables substitution
+MANIFESTS_DIR=output
```

Comment:
`MANIFESTS_DIR=$(mktemp -d) || (echo "Failed to create temp directory" && exit 1)`
and use trap to clean it